conan/internal/runner/docker.py

```diff
@@ -0,0 +1,268 @@
+                build_args=_instans_or_error(runnerfile.get('build', {}).get('build_args'), dict),
+                cache_from=_instans_or_error(runnerfile.get('build', {}).get('cacheFrom'), list),
+            ),
+            run=Run(
+                name=_instans_or_error(runnerfile.get('run', {}).get('name'), str),
+                environment=_instans_or_error(runnerfile.get('run', {}).get('containerEnv'), dict),
+                user=_instans_or_error(runnerfile.get('run', {}).get('containerUser'), str),
+                privileged=_instans_or_error(runnerfile.get('run', {}).get('privileged'), bool),
+                cap_add=_instans_or_error(runnerfile.get('run', {}).get('capAdd'), list),
+                security_opt=_instans_or_error(runnerfile.get('run', {}).get('securityOpt'), list),
+                volumes=_instans_or_error(runnerfile.get('run', {}).get('mounts'), dict),
+            )
+        )
+    else:
+        return Conf(
+            image=None,
+            build=Build(dockerfile=None, build_context=None, build_args=None, cache_from=None),
+            run=Run(name=None, environment=None, user=None, privileged=None, cap_add=None, security_opt=None, volumes=None)
+        )
+
+
+def _docker_info(msg, error=False):
+    fg=Color.BRIGHT_MAGENTA
+    if error:
+        fg=Color.BRIGHT_RED
+    ConanOutput().status('\n┌'+'─'*(2+len(msg))+'┐', fg=fg)
+    ConanOutput().status(f'| {msg} |', fg=fg)
+    ConanOutput().status('└'+'─'*(2+len(msg))+'┘\n', fg=fg)
+
+
+class DockerRunner:
+    def __init__(self, conan_api, command, host_profile, build_profile, args, raw_args):
+        import docker
+        import docker.api.build
+        try:
+            self.docker_client = docker.from_env()
+            self.docker_api = docker.APIClient()
+            docker.api.build.process_dockerfile = lambda dockerfile, path: ('Dockerfile', dockerfile)
+        except:
+            raise ConanException("Docker Client failed to initialize."
+                                 "\n - Check if docker is installed and running"
+                                 "\n - Run 'pip install pip install conan[runners]'")
+        self.conan_api = conan_api
+        self.build_profile = build_profile
+        self.args = args
+        self.abs_host_path = make_abs_path(args.path)
+        if args.format:
+            raise ConanException("format argument is forbidden if running in a docker runner")
+
+        # Runner config
+        self.abs_runner_home_path = os.path.join(self.abs_host_path, '.conanrunner')
+        self.abs_docker_path = os.path.join('/root/conanrunner', os.path.basename(self.abs_host_path)).replace("\\","/")
+
+        # Update conan command and some paths to run inside the container
+        raw_args[raw_args.index(args.path)] = self.abs_docker_path
+        self.profiles = []
+        profile_list = set(self.args.profile_build + self.args.profile_host)
```

Comment:
When I run `conan create . -pr=docker_linux_runner` (So no build profile) this throws with an exception. I know you _have_ to define the `build` profile as it can't just be the default, but the `default` is only added by the profile load function call in create, so that `self.args.build_profile` will still be None if nothing is passed